src/main/java/uk/gov/companieshouse/insolvency/delta/validation/InsolvencyDeltaValidator.java

```diff
@@ -70,4 +73,25 @@ private List<String> buildListOfFieldsToValidateForSchema(
                 Predicate.not(caseDates.get(caseNumber.getCaseTypeId().getValue())::contains))
                 .collect(Collectors.toList());
     }
+
+    private void checkForMandatoryCaseDate(CaseNumber caseNumber,
+                                    Map<Integer, String> mandatoryCaseTypesField) throws Exception {
+
+        int caseTypeId = caseNumber.getCaseTypeId().getValue();
+
+        if (mandatoryCaseTypesField.containsKey(caseTypeId)) {
+            if (!checkIfFieldIsPresent(mandatoryCaseTypesField.get(caseTypeId), caseNumber)) {
```

Comment:
You could do this in one if check like: 
```suggestion
        if (mandatoryCaseTypesField.containsKey(caseTypeId)
            && !checkIfFieldIsPresent(mandatoryCaseTypesField.get(caseTypeId), caseNumber)) {
```